unblob/processing.py

```diff
@@ -149,13 +176,15 @@ def _process_task(  # noqa: C901
                 calculate_entropy(carved_path, draw_plot=config.verbose)
 
         for chunk in outer_chunks:
-            new_path = extract_valid_chunk(extract_dir, file, chunk)
-            task_queue.put(
-                Task(
-                    root=config.extract_root,
-                    path=new_path,
-                    depth=task.depth + 1,
-                )
+            new_path = extract_valid_chunk(extract_dir, file, chunk, result_queue)
+
+            if new_path is None:
```

Comment:
This makes no sense to me. extract_valid_chunk should always extract a ValidChunk and the code should be at a different place deciding if it's indeed valid or not.